`test_delta_from_def_2` fails on aarch64, armv7 and ppc64le

**Describe the issue**:
`test_delta_from_def_2` fails on aarch64, armv7 and ppc64le.

```
[  371s] =================================== FAILURES ===================================
[  371s] ____________________________ test_delta_from_def_2 _____________________________
[  371s] [gw1] linux -- Python 3.8.15 /usr/bin/python3.8
[  371s] 
[  371s]     def test_delta_from_def_2():
[  371s]         # https://github.com/apache/parquet-format/blob/master/Encodings.md#example-2
[  371s]         expected = [7, 5, 3, 1, 2, 3, 4, 5]
[  371s]         data = np.zeros(1000, dtype="uint8")
[  371s]         o = cencoding.NumpyIO(data)
[  371s]         cencoding.encode_unsigned_varint(8, o)  # block size
[  371s]         cencoding.encode_unsigned_varint(1, o)  # miniblock count
[  371s]         cencoding.encode_unsigned_varint(8, o)  # value count
[  371s]         cencoding.encode_unsigned_varint(zigzag(7), o)  # first value (zigzag)
[  371s]     
[  371s]         # one and only miniblock
[  371s]         cencoding.encode_unsigned_varint(zigzag(-2), o)  # minimum delta (zigzag)
[  371s]         o.write_byte(2)  # bit-width list (only one)
[  371s]         o.write_byte(0b00000011)  # [0, 0, 0, 3]
[  371s]         o.write_byte(0b11111100)  # [3, 3, 3, pad]
[  371s]     
[  371s]         o.seek(0)
[  371s]     
[  371s]         outdata = np.zeros(1000, dtype="int32")
[  371s]         out = cencoding.NumpyIO(outdata.view("uint8"))
[  371s]         cencoding.delta_binary_unpack(o, out)
[  371s] >       assert outdata[:8].tolist() == expected
[  371s] E       assert [7, 5, 3, 1, -1, -3, -5, -7] == [7, 5, 3, 1, 2, 3, 4, 5]
[  371s] E         At index 4 diff: -1 != 2
[  371s] E         Full diff:
[  371s] E         - [7, 5, 3, 1, 2, 3, 4, 5]
[  371s] E         ?              ^     ^  ^
[  371s] E         + [7, 5, 3, 1, -1, -3, -5, -7]
[  371s] E         ?              ^^  +   ^^  ^^
[  371s] 
[  371s] ../../BUILDROOT/python-fastparquet-0.8.3-1.1.aarch64/usr/lib64/python3.8/site-packages/fastparquet/test/test_encoding.py:176: AssertionError
```

**Minimal Complete Verifiable Example**:

```python
# Put your MCVE code here
```

**Anything else we need to know?**:

**Environment**:

- Dask version:
- Python version: 3.8
- Operating System: openSUSE Tumbleweed
- Install method (conda, pip, source): source

Tested 2022.12.0:
```
[  175s] =================================== FAILURES ===================================
[  175s] ____________________________ test_delta_from_def_2 _____________________________
[  175s] [gw0] linux -- Python 3.8.16 /usr/bin/python3.8
[  175s] 
[  175s]     def test_delta_from_def_2():
[  175s]         # https://github.com/apache/parquet-format/blob/master/Encodings.md#example-2
[  175s]         expected = [7, 5, 3, 1, 2, 3, 4, 5]
[  175s]         data = np.zeros(1000, dtype="uint8")
[  175s]         o = cencoding.NumpyIO(data)
[  175s]         cencoding.encode_unsigned_varint(8, o)  # block size
[  175s]         cencoding.encode_unsigned_varint(1, o)  # miniblock count
[  175s]         cencoding.encode_unsigned_varint(8, o)  # value count
[  175s]         cencoding.encode_unsigned_varint(zigzag(7), o)  # first value (zigzag)
[  175s]     
[  175s]         # one and only miniblock
[  175s]         cencoding.encode_unsigned_varint(zigzag(-2), o)  # minimum delta (zigzag)
[  175s]         o.write_byte(2)  # bit-width list (only one)
[  175s]         o.write_byte(0b00000011)  # [0, 0, 0, 3]
[  175s]         o.write_byte(0b11111100)  # [3, 3, 3, pad]
[  175s]     
[  175s]         o.seek(0)
[  175s]     
[  175s]         outdata = np.zeros(1000, dtype="int32")
[  175s]         out = cencoding.NumpyIO(outdata.view("uint8"))
[  175s]         cencoding.delta_binary_unpack(o, out)
[  175s] >       assert outdata[:8].tolist() == expected
[  175s] E       assert [7, 5, 3, 1, -1, -3, -5, -7] == [7, 5, 3, 1, 2, 3, 4, 5]
[  175s] E         At index 4 diff: -1 != 2
[  175s] E         Full diff:
[  175s] E         - [7, 5, 3, 1, 2, 3, 4, 5]
[  175s] E         ?              ^     ^  ^
[  175s] E         + [7, 5, 3, 1, -1, -3, -5, -7]
[  175s] E         ?              ^^  +   ^^  ^^
[  175s] 
```

--- a/fastparquet/cencoding.pyx
+++ b/fastparquet/cencoding.pyx
@@ -18,7 +18,7 @@ cdef extern from "string.h":
 from cpython cimport (
     PyBytes_FromStringAndSize, PyBytes_GET_SIZE, PyUnicode_DecodeUTF8,
 )
-from libc.stdint cimport uint8_t, uint32_t, int32_t, uint64_t, int64_t
+from libc.stdint cimport int8_t, uint8_t, uint32_t, int32_t, uint64_t, int64_t
 
 
 cpdef void read_rle(NumpyIO file_obj, int32_t header, int32_t bit_width, NumpyIO o, int32_t itemsize=4):
@@ -217,7 +217,7 @@ cdef void delta_read_bitpacked(NumpyIO file_obj, uint8_t bitwidth,
                                NumpyIO o, uint64_t count, uint8_t itemsize=4):
     cdef:
         uint64_t data = 0
-        char stop = -bitwidth
+        int8_t stop = -bitwidth
         uint64_t mask = 0XFFFFFFFFFFFFFFFF >> (64 - bitwidth)
     while count > 0:
         if stop < 0:

--- a/fastparquet/cencoding.pyx
+++ b/fastparquet/cencoding.pyx
@@ -225,6 +225,7 @@ cdef void delta_read_bitpacked(NumpyIO file_obj, uint8_t bitwidth,
             stop += 8
         else:
             o.write_int((data >> stop) & mask)
+            print("bitpack value", (data >> stop) & mask, data, stop, mask)
             stop -= bitwidth
             count -= 1
 
@@ -239,11 +240,13 @@ cpdef void delta_binary_unpack(NumpyIO file_obj, NumpyIO o):
         const uint8_t[:] bitwidths
         uint8_t bitwidth
     values_per_miniblock = block_size // miniblock_per_block
+    print("\nstart", count, value, values_per_miniblock)
     while True:
         min_delta = zigzag_long(read_unsigned_var_int(file_obj))
         bitwidths = file_obj.read(miniblock_per_block)
         for i in range(miniblock_per_block):
             bitwidth = bitwidths[i]
+            print("\n  miniblock", i, "width", bitwidth)
             if bitwidth:
                 temp = o.loc
                 if count > 1:
@@ -253,6 +256,7 @@ cpdef void delta_binary_unpack(NumpyIO file_obj, NumpyIO o):
                 for j in range(values_per_miniblock):
                     temp = o.read_int()
                     o.loc -= 4
+                    print("miniblock value", value)
                     o.write_int(value)
                     value += min_delta + temp
                     count -= 1
@@ -260,6 +264,7 @@ cpdef void delta_binary_unpack(NumpyIO file_obj, NumpyIO o):
                         return
             else:
                 for j in range(values_per_miniblock):
+                    print("miniblock value", value)
                     o.write_int(value)
                     value += min_delta
                     count -= 1

--- a/fastparquet/cencoding.pyx
+++ b/fastparquet/cencoding.pyx
@@ -223,10 +223,8 @@ cdef void delta_read_bitpacked(NumpyIO file_obj, uint8_t bitwidth,
         if stop < 0:
             data = ((data & 0X00FFFFFFFFFFFFFF) << 8) | file_obj.read_byte()
             stop += 8
-            print("bin stop", bin(stop), bin(data))
         else:
             o.write_int((data >> stop) & mask)
-            print("bitpack value", (data >> stop) & mask, data, stop, mask)
             stop -= bitwidth
             count -= 1
 
@@ -241,13 +239,11 @@ cpdef void delta_binary_unpack(NumpyIO file_obj, NumpyIO o):
         const uint8_t[:] bitwidths
         uint8_t bitwidth
     values_per_miniblock = block_size // miniblock_per_block
-    print("\nstart", count, value, values_per_miniblock)
     while True:
         min_delta = zigzag_long(read_unsigned_var_int(file_obj))
         bitwidths = file_obj.read(miniblock_per_block)
         for i in range(miniblock_per_block):
             bitwidth = bitwidths[i]
-            print("\n  miniblock", i, "width", bitwidth)
             if bitwidth:
                 temp = o.loc
                 if count > 1:
@@ -257,7 +253,6 @@ cpdef void delta_binary_unpack(NumpyIO file_obj, NumpyIO o):
                 for j in range(values_per_miniblock):
                     temp = o.read_int()
                     o.loc -= 4
-                    print("miniblock value (bw)", value)
                     o.write_int(value)
                     value += min_delta + temp
                     count -= 1
@@ -265,7 +260,6 @@ cpdef void delta_binary_unpack(NumpyIO file_obj, NumpyIO o):
                         return
             else:
                 for j in range(values_per_miniblock):
-                    print("miniblock value", value)
                     o.write_int(value)
                     value += min_delta
                     count -= 1

--- a/fastparquet/cencoding.pyx
+++ b/fastparquet/cencoding.pyx
@@ -223,6 +223,7 @@ cdef void delta_read_bitpacked(NumpyIO file_obj, uint8_t bitwidth,
         if stop < 0:
             data = ((data & 0X00FFFFFFFFFFFFFF) << 8) | file_obj.read_byte()
             stop += 8
+            print("bin stop", bin(stop), bin(data))
         else:
             o.write_int((data >> stop) & mask)
             print("bitpack value", (data >> stop) & mask, data, stop, mask)

--- a/fastparquet/cencoding.pyx
+++ b/fastparquet/cencoding.pyx
@@ -256,7 +256,7 @@ cpdef void delta_binary_unpack(NumpyIO file_obj, NumpyIO o):
                 for j in range(values_per_miniblock):
                     temp = o.read_int()
                     o.loc -= 4
-                    print("miniblock value", value)
+                    print("miniblock value (bw)", value)
                     o.write_int(value)
                     value += min_delta + temp
                     count -= 1

